### PR TITLE
fix Sympy Lambdify error when expr atoms have brace

### LINKED ISSUE #23374: Lambdify with latex symbols
I have a complicated expression that needs to be evaluated. If the evaluation fails with numpy, than it switches to sympy. My symbols contains latex syntax.

This is the simplest reproducible example I could come up with:

```python
from sympy import *
import inspect
x1, x2 = symbols("x_{1} x_2")

f1 = lambdify([x1, x2], cos(x1**2 + x2**2))
print(inspect.getsource(f1))

# def _lambdifygenerated(x_1, x_2):
#     return cos(x_2**2 + x_1**2) 
```
So far so good. Note that `x1` was processed to `x_1`. Now, let's switch to SymPy:

```python
f2 = lambdify([x1, x2], cos(x1**2 + x2**2), modules="sympy")
print(inspect.getsource(f2))
```

```text
Traceback (most recent call last):

  File "/home/davide/Documents/Development/envs/sympy_plot/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3457, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)

  File "/tmp/ipykernel_82217/2465285261.py", line 8, in <module>
    f2 = lambdify([x1, x2], cos(x1**2 + x2**2), modules="sympy")

  File "/home/davide/Documents/Development/envs/sympy_plot/lib/python3.9/site-packages/sympy/utilities/lambdify.py", line 900, in lambdify
    c = compile(funcstr, filename, 'exec')

  File "<lambdifygenerated-12>", line 2
    return cos(x_2**2 + x_{1}**2)
                          ^
SyntaxError: invalid syntax
```

Note that no processing is applied to `x1`, so curl brackets end up in the expression to be executed.

### PR DESCRIPTION
**Pull request by cocolato**
#### References to other Issues or PRs

Fix #23374

#### Brief description of what is fixed or changed
In #20806, When use `PythonCodePrinter` to print a `Symbol` object will remove curly braces.
https://github.com/sympy/sympy/blob/ab414d124bb9899eb9d677e95498c7a021c9bc29/sympy/printing/pycode.py#L564-L577

In `Sympy Lambdify` will use  `LambdaPrinter()` to print args but use `SymPyPrinter` to print the expr. The `SymPyPrinter` don't have the special `_print_Symbol` function to remove curly braces.

https://github.com/sympy/sympy/blob/ab414d124bb9899eb9d677e95498c7a021c9bc29/sympy/utilities/lambdify.py#L1114-L1115

https://github.com/sympy/sympy/blob/ab414d124bb9899eb9d677e95498c7a021c9bc29/sympy/utilities/lambdify.py#L1148-L1168

#### Release Notes

* printing
  * Remove symbol's curly braces in `SymPyPrinter`.

### CODE REVIEW

**Comment by cocolato** [comment]
@oscargus Hi, can you review this pull request? If you have other suggestions, I'll try commit more changes.

**Comment by oscargus** [comment]
I believe that this should be applied in codeprinter instead (as pointed out in https://github.com/sympy/sympy/issues/23374#issuecomment-1124788078) as, AFAIK, using {} in variable names is not really supported by most languages.

This also means that the specific `_print_Symbol` methods can be removed from some of the other printers (as pointed out in https://github.com/sympy/sympy/issues/23374#issuecomment-1124792448).

**Comment by cocolato** [comment]
> I believe that this should be applied in codeprinter instead (as pointed out in [#23374 (comment)](https://github.com/sympy/sympy/issues/23374#issuecomment-1124788078)) as, AFAIK, using {} in variable names is not really supported by most languages.
> 
> This also means that the specific `_print_Symbol` methods can be removed from some of the other printers (as pointed out in [#23374 (comment)](https://github.com/sympy/sympy/issues/23374#issuecomment-1124792448)).

Ok，just remove the specific `_print_Symbol` from `PythonCodePrinter`?  But how to deal with the problem in  #20762 ?

**Comment by cocolato** [comment]
> If you put it in `codeprinter.py`, it will inherit from there (and all other languages will also benefit from this).

Thanks! And I wonder if the current commit is as expected?

File: sympy/printing/pycode.py [hunk 1]
```diff
@@ -571,8 +571,6 @@ def _print_Symbol(self, expr):
                        'reserved keyword in this language.')
                 raise ValueError(msg.format(name))
             return name + self._settings['reserved_word_suffix']
-        elif '{' in name:   # Remove curly braces from subscripted variables
```
Comment (oscargus) [inline sympy/printing/pycode.py:574]: It should be possible to delete this method completely now. (And maybe some other identical ones if there are any.)

**Comment by oscargus** [comment]
Except for removing the, now, redundant method, this looks good!

**Comment by asmeurer** [comment]
Shouldn't this be handled by the `dummify` flag to `lambdify`?

**Comment by oscargus** [comment]
Basically, I think we get more bug reports from people using {} in variable names and then something else breaks than we would get bug reports from people being upset with that we do not generate code where they are still there (as that code will be syntactically incorrect anyway).

File: sympy/printing/pycode.py (hunk 1 above)
Comment (oscargus) [reply sympy/printing/pycode.py:574]: I mean: you should remove the whole `_print_Symbol`-method here as it will be inherited from `codeprinter` now.

Comment (cocolato) [reply sympy/printing/pycode.py:574]: I see, thanks！

**Comment by oscargus** [comment]
Thanks! 

I seems like the failing tests may be related (since they are code-printing tests). However, as they only fail on pypy and Python 3.11 it is not obvious what may be wrong. I'll restart those tests and see what happens...

**Comment by asmeurer** [comment]
Ideally this shouldn't even be needed. dummify ought to take care of it automatically (see https://github.com/sympy/sympy/issues/12463). 

I guess you could argue that the problem isn't solved for people using the printer directly. Maybe the logic should be unified so that it works with both. Although one could argue that if you are using the printers directly it is your responsibility to use safe names for your symbols.

Either way, there are a lot more characters than just braces that would break in symbol names.

**Comment by asmeurer** [comment]
Also, *removing* braces could lead to a clash, e.g., if `x{1}` and `x1` both exist and are distinct things. dummify doesn't have this problem. Handling this sort of thing in the printer requires a bit more thought than this, IMO.

**Comment by oscargus** [comment]
> Either way, there are a lot more characters than just braces that would break in symbol names.

Agreed. It seems like {} is the most common cause for issues though (as people tend to use them to get nice latex-printing).

Not sure what the right way to go here is, but I am quite sure that I have fixed similar bugs in the past, one printer at a time...

Regarding clashes: another topic to mention in the "naming symbols tutorial"... 

(I wasn't aware of dummify, is there a reason that is not on by default?)

**Pull request closed without merging**